postgresql/resource_postgresql_grant.go

```diff
@@ -730,6 +730,8 @@ func checkRoleDBSchemaExists(client *Client, d *schema.ResourceData) (bool, erro
 		return false, nil
 	}
 
+	deferredRollback(txn)
```

Comment:
I think we should better avoid creating a new transaction and use the existing one if it's the same database.

So line 738:

```go
var dbTxn *sql.Tx
if database == client.databaseName {
  dbTxn = txn
} else {
	var err error;
	dbTxn, err := startTransaction(client, database)
	if err != nil {
		return false, err
	}
	defer deferredRollback(dbTxn)
}
```

(and actually if it's for another database a transaction is not really needed as it executes only 1 request but I don't think we already have an helper to get a simple DB connection for alternative DB)

What do you think?

Reply:
Makes sense to me. I'll unify this function to be more in line with the rest of the code.